src/pybind/mgr/dashboard/controllers/multi_cluster.py

```diff
@@ -402,12 +549,49 @@ class MultiClusterUi(RESTController):
     def set_cors_endpoint(self, url: str):
         configure_cors(url)
 
+    @Endpoint('GET')
+    @ReadPermission
+    def get_prometheus_access_info(self):
+        orch_backend = mgr.get_module_option_ex('orchestrator', 'orchestrator')
+        if orch_backend == 'cephadm':
+            cmd = {
+                'prefix': 'orch prometheus get-credentials',
+            }
+            ret_status, out, _ = mgr.mon_command(cmd)
+            if ret_status == 0 and out is not None:
+                prom_access_info = json.loads(out)
+                user = prom_access_info['user']
+                password = prom_access_info['password']
+                certificate = prom_access_info['certificate']
+            return {
+                'user': user,
+                'password': password,
+                'certificate': certificate
+            }
+        return None
+
 
 def _set_prometheus_targets(prometheus_url: str):
     orch_backend = mgr.get_module_option_ex('orchestrator', 'orchestrator')
-    if orch_backend == 'cephadm':
-        cmd = {
-            'prefix': 'orch prometheus set-target',
-            'url': prometheus_url.replace('http://', '').replace('https://', '')
-        }
-        mgr.mon_command(cmd)
+    try:
+        if orch_backend == 'cephadm':
+            cmd = {
+                'prefix': 'orch prometheus set-target',
+                'url': prometheus_url.replace('http://', '').replace('https://', '')
```

Comment:
... we are calling url something which is not. Please, we have to be precise. This seems to be an endpoint not a URL